src/rf/apps/home/views.py

```diff
@@ -3,23 +3,154 @@
 from __future__ import unicode_literals
 from __future__ import division
 
+from django.shortcuts import (get_object_or_404,
+                              render_to_response)
 from django.contrib.auth.models import User
-from django.shortcuts import render_to_response
-from rest_framework import serializers, viewsets
 
+from rest_framework import (viewsets, mixins,
+                            status, generics,
+                            filters)
+from rest_framework.decorators import detail_route
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
 
-# Serializers define the API representation.
-class UserSerializer(serializers.HyperlinkedModelSerializer):
-    class Meta:
-        model = User
-        fields = ('url', 'username', 'email', 'is_staff')
-
+import django_filters
 
-# ViewSets define the view behavior.
-class UserViewSet(viewsets.ModelViewSet):
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
+from apps.core.models import Layer, LayerMeta, UserFavoriteLayer
+from apps.core.serializers import LayerSerializer, LayerMetaSerializer
 
 
 def home_page(request):
     return render_to_response('home/home.html')
+
+
+class LayerFilter(django_filters.FilterSet):
+    tag = django_filters.CharFilter(name='layer_tags__name')
+
+    class Meta:
+        model = Layer
+        fields = ('name', 'organization', 'tag')
+
+
+# This class is derived from ViewSet (instead of ApiView) because
+# it automatically routes /user/<username>/layers/<slug>/
+# to destroy and retrieve, and /user/<username>/layers/
+# to create and list.
+class UserLayerViewSet(mixins.ListModelMixin,
+                       mixins.RetrieveModelMixin,
+                       mixins.CreateModelMixin,
+                       mixins.DestroyModelMixin,
+                       viewsets.GenericViewSet):
+    serializer_class = LayerSerializer
+    lookup_field = 'slug'
+    permission_classes = [IsAuthenticated]
+
+    filter_backends = (filters.DjangoFilterBackend, filters.OrderingFilter,)
+    filter_class = LayerFilter
+    ordering_fields = ('name', 'organization')
+
+    def get_queryset(self):
+        get_object_or_404(User, username=self.kwargs['username'])
+        if self.request.user.username != self.kwargs['username']:
+            return Layer.objects.filter(user__username=self.kwargs['username'],
+                                        is_public=True)
+        else:
+            return Layer.objects.filter(user__username=self.kwargs['username'])
+
+    def retrieve(self, request, *args, **kwargs):
+        if (request.user.username != kwargs['username'] and not
+                self.get_object().is_public):
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Could possibly create a custom DRF permission class to reduce this boilerplate. (Ref: https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/permissions.py#L42)